src/springsettings.js

```diff
@@ -7,58 +7,81 @@ const os = require('os');
 const { writePath } = require('./spring_platform');
 
 class Springsettings extends EventEmitter {
-	applyDefaultsOnSettings(defaults, springsettingsPath) {
+	// Parsing according to implementation in rts/System/Config/ConfigSource.cpp
```

Comment:
Let's first discuss this single comment before others, because based on the result of discussion here I might need to change a lot, and a lot of other comments might be just obsolete.

The reasons why I'm using the engine implementation is:
1. That's what in the end is using and rewriting the config on load anyway so I thought that's a reasonable idea. With this, I don't need to think how to treat user config, but just follow what is already there.
2. It has easy separation of parsing, making operations on config, and writing: which is much easier to reason about then let's read, parse, change and write in the same loop. It also doesn't need any additional data structures to store the config, it's just a map.

If you prefer to do something smarter then spring engine, and hope that spring engine will also get smarter in handling it, I can rewrite this. If we keep it as spring engine, I will push back on a bunch of comments with reason "that's what spring engine does".

Your call, I don't mind.

Reply:
I think a good argument is: If spring engine improves config handling, then spring launcher will have this broken behavior.

It's a bit hard to judge in which direction spring engine would improve config parsing though, but, there are some thing that can be done irrespectively: e.g. don't sort lines.

Reply:
Admittedly you make a fair point: even if spring-launcher doesn't preserve the original config structure, spring is likely to mess it up anyway, so there isn't much of a point in us trying to preserve user configs. 

That said, I don't think this lends itself to code simplicity. I think the `comments` part is particularly confusing (comments having *keys* and there being an `endComment` thing). Separating parsing, config in-memory editing and writing is a good thing, but I think that can be done while still preserving the original order. 

I think a data structure for in-memory config representation (that supports edit operations and serialization to & from files/strings) would solve my concerns and still allow us to have a good separation of config handling stages. 
Would you be interested in making it? I also understand this might not be a high priority due to the reasons mentioned in the first paragraph.

Reply:
I will take a stab at making something readable after I finish my workday ;), let's see how it plays out.

Reply:
Done, resolving.